aptos-move/aptos-vm/src/zkid_validation.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::move_vm_ext::{AptosMoveResolver, SessionExt};
+use anyhow::{Context, Result};
+use aptos_types::{
+    jwks::{jwk::JWK, PatchedJWKs},
+    on_chain_config::{CurrentTimeMicroseconds, OnChainConfig},
+    transaction::SignedTransaction,
+    vm_status::{StatusCode, VMStatus},
+    zkid::{ZkpOrOpenIdSig, MAX_ZK_ID_AUTHENTICATORS_ALLOWED},
+};
+use aptos_vm_logging::log_schema::AdapterLogSchema;
+use move_core_types::{language_storage::CORE_CODE_ADDRESS, move_resource::MoveStructType};
+
+fn get_current_time_onchain(resolver: &impl AptosMoveResolver) -> Result<CurrentTimeMicroseconds> {
```

Comment:
For configs, you can do something like `CurrentTimeMicroseconds::fetch_config(resolver)`. That trait is purely for configs but returns an Option, so you would need to map error